lib/spack/spack/build_systems/cmake.py

```diff
@@ -355,6 +355,18 @@ def std_cmake_args(self):
         args += getattr(self.pkg, "cmake_flag_args", [])
         return args
 
+    @staticmethod
+    def setup_dependent_cmake_package(module, dependent_spec, *propegated_args):
+        """Helper method to propegate required CMake cache definitions to dependent
+        packages"""
+        if CMakeBuilder in type(dependent_spec.package.builder).__mro__:
```

Comment:
Everything that inherits `CMakePackage` will include a `build_system=cmake`, which I think is more robust than this check.

Reply:
I think they're effectively equivalent (if a package inherits from `CMakePackage` its builder will inherit from `CMakeBuilder`) but I agree the spec approach is less visually jarring/looks less hacky.